spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -0,0 +1,461 @@
+
+      this.delegate = new ClusteredPositionDeleteWriter<>(
+          writerFactory, deleteFileFactory, io,
+          ctx.deleteFileFormat(), ctx.targetDeleteFileSize());
+      this.positionDelete = PositionDelete.create();
+      this.io = io;
+      this.specs = specs;
+      this.partitionRowWrapper = partitionRowWrapper;
+      this.partitionProjections = partitionProjections;
+      this.specIdOrdinal = ctx.metadataSparkType().fieldIndex(MetadataColumns.SPEC_ID.name());
+      this.partitionOrdinal = ctx.metadataSparkType().fieldIndex(MetadataColumns.PARTITION_COLUMN_NAME);
+      this.fileOrdinal = ctx.deleteSparkType().fieldIndex(MetadataColumns.FILE_PATH.name());
+      this.positionOrdinal = ctx.deleteSparkType().fieldIndex(MetadataColumns.ROW_POSITION.name());
+    }
+
+    @Override
+    public void delete(InternalRow meta, InternalRow id) throws IOException {
+      int specId = meta.getInt(specIdOrdinal);
+      PartitionSpec spec = specs.get(specId);
+
+      InternalRow partition = meta.getStruct(partitionOrdinal, partitionRowWrapper.size());
+      StructProjection partitionProjection = partitionProjections.get(specId);
+      partitionProjection.wrap(partitionRowWrapper.wrap(partition));
+
+      String file = id.getString(fileOrdinal);
+      long position = id.getLong(positionOrdinal);
+      positionDelete.set(file, position, null);
+      delegate.write(positionDelete, spec, partitionProjection);
+    }
+
+    @Override
+    public void update(InternalRow metadata, InternalRow id, InternalRow row) {
+      throw new UnsupportedOperationException(this.getClass().getName() + " does not implement update");
+    }
+
+    @Override
+    public void insert(InternalRow row) throws IOException {
+      throw new UnsupportedOperationException(this.getClass().getName() + " does not implement insert");
+    }
+
+    @Override
+    public WriterCommitMessage commit() throws IOException {
+      close();
+
+      DeleteWriteResult result = delegate.result();
+      return new DeltaTaskCommit(result);
+    }
+
+    @Override
+    public void abort() throws IOException {
+      close();
+
+      DeleteWriteResult result = delegate.result();
+      cleanFiles(io, result.deleteFiles());
+    }
+
+    @Override
+    public void close() throws IOException {
+      if (!closed) {
+        delegate.close();
+        this.closed = true;
+      }
+    }
+  }
+
+  private static class Context implements Serializable {
+    private final Schema dataSchema;
+    private final StructType dataSparkType;
+    private final FileFormat dataFileFormat;
+    private final long targetDataFileSize;
+    private final StructType deleteSparkType;
+    private final StructType metadataSparkType;
+    private final FileFormat deleteFileFormat;
+    private final long targetDeleteFileSize;
+    private final boolean fanoutWriterEnabled;
+
+    Context(Schema dataSchema, SparkWriteConf writeConf, ExtendedLogicalWriteInfo info) {
```

Comment:
I'm not sure how much this class actually helps because most of what it does is returning values that are from `info` and `writeConf`. Is the value of this that the `SparkWriteConf` is not `Serializable`? If so, I think it would be good to note that in the class's javadoc.

Reply:
I am open to alternatives. The reason why I added it is to avoid passing 9 arguments separately. I usually try to avoid huge arg lists that need a few lines. That's a personal preference, though. What do you think, @rdblue?

Reply:
The arguments don't need to be passed in, you can just pass these 3 and set instance fields from them.

Reply:
You are right it has to be serialized and `SparkWriteConf` isn't. That's probably why I added it.

Reply:
Added a comment at the top.